### ISSUE TITLE
Do not send data when running on user devices

### ISSUE THREAD

**Issue by Bastian** [Bastian/bstats-metrics#113]
Sending data from user devices without their explicit consent (i.e., opt in) probably violates the GDPR. Therefore the Metrics class should detect if it runs on a user device and not send data in these cases.

See https://github.com/Bastian/bstats-metrics/issues/112#issuecomment-1365340696

**Comment by Pingger** [comment]
That just moves the issue. And how can you guarantee, that the device is not a "user" device? There are quite a few simple tools, that automatically make use of containers, that are widely in use. This often makes it effectively impossible to distinguish between "user" and "server" devices.

**Comment by KevinDaGame** [comment]
Detecting if it is a user device is not a good option in my opinion since there is just too big an error margin (I personally use Linux as an OS for example). I do agree that something should be put in place. (Perhabs a popup in chat to OPS that shows until either agreed or denied?)

**Comment by Bastian** [comment]
Yes, some edge cases like containers might remain, but some simple checks should cover most edge cases. I'm open for suggestions for impovements.

**Comment by KevinDaGame** [comment]
Another point which a friend pointed out: Right now, BStats will send metrics when ran on localhost servers that are not port forwarded. This would be a case where there is (as I see it) no argument to be made about "intentionally publishing your ip" so that should definitely be a check in BStats in my opinion
